src/librustc_metadata/loader.rs

```diff
@@ -342,6 +343,7 @@ impl<'a> Context<'a> {
                              "found crate `{}` compiled by an incompatible version of rustc{}",
                              self.ident, add)
         } else {
+            err_msg = format!("can't find crate");
             struct_span_err!(self.sess, self.span, E0463,
                              "can't find crate for `{}`{}",
                              self.ident, add)
```

Comment:
Rather than setting err_msg and using it later, you could instead set the label in the same condition.  Maybe something like:

```rust
             let mut err = struct_span_err!(self.sess, self.span, E0463,
                                            "can't find crate for `{}`{}",
                                            self.ident, add);
             err.span_label(self.span, &format!("can't find crate"));
             err
```